File: tests/fixtures/real_world/python/taint/closure_captured_var.py
# Python: nested def captures outer var, then inner() is called.
import os
from flask import Flask, request

app = Flask(__name__)

@app.route('/a')
def handler():
    q = request.args.get('q')   # source
    def inner():
        os.system(q)            # sink on captured source
    inner()
    return 'ok'

File: tests/fixtures/real_world/javascript/taint/closure_captured_var.js
// Closure captures a tainted outer var and is called later in same scope.
// Expected: sink triggers on req.query.q → eval inside closure.
var express = require('express');
var app = express();

app.get('/a', function(req, res) {
    var q = req.query.q;                 // source
    var fn = function() { eval(q); };   // closure captures q
    fn();                                // invoke captured closure
    res.send('ok');
});
